dpnp/backend/src/dpnp_utils.hpp

```diff
@@ -228,6 +230,30 @@ static inline bool
     return std::equal(std::begin(input1_vec), std::end(input1_vec), std::begin(input2_vec));
 }
 
+/**
+ * @ingroup BACKEND_UTILS
+ * @brief Cast vector of DPCtl events to vector of SYCL enents.
+ *
+ * @param [in] events_ref      Reference to vector of DPCtl events.
+ *
+ * @return                     Vector of SYCL events.
+ */
+static inline std::vector<sycl::event>
```

Comment:
Instead of using `static inline`, put the function into anonymous namespace to ensure code inlining:

```
namespace{
   std::vector<sycl::event> cast_event_vector(..) { ... }
}

Reply:
Per https://en.cppreference.com/w/cpp/language/inline the `inline` in C++ does not mean code inline, means multiple definitions are permitted (https://en.cppreference.com/w/cpp/language/inline).